vsintegration/src/FSharp.Editor/Navigation/GoToDefinition.fs

```diff
@@ -726,4 +728,125 @@ type internal FSharpNavigation
         
             // Don't show the dialog box as it's most likely that the user cancelled.
             // Don't make them click twice.
-            true
\ No newline at end of file
+            true
+
+type private SymbolPath = { EntityPath: string list; MemberOrValName: string }
+[<RequireQualifiedAccess>]
+type private DocCommentId =
+    | Member of SymbolPath
+    | Type of EntityPath: string list
+    | Other of string
+    | None
+    
+
+type FSharpNavigableLocation(statusBar: StatusBar, metadataAsSource: FSharpMetadataAsSourceService, symbolRange: range, project: Project) =
+    interface IFSharpNavigableLocation with
+        member _.NavigateToAsync(_options: FSharpNavigationOptions2, cancellationToken: CancellationToken) : Task<bool> =
+            asyncMaybe {
+                let targetPath = symbolRange.FileName
+                let! targetDoc = project.Solution.TryGetDocumentFromFSharpRange (symbolRange, project.Id)
+                let! targetSource = targetDoc.GetTextAsync(cancellationToken)
+                let gtd = GoToDefinition(metadataAsSource)
+
+                let (|Signature|Implementation|) filepath =
+                    if isSignatureFile filepath then Signature else Implementation
+                
+                match targetPath with
+                | Signature  ->
+                    return! gtd.NavigateToSymbolDefinitionAsync(targetDoc, targetSource, symbolRange, statusBar, cancellationToken)
+                | Implementation ->
+                    return! gtd.NavigateToSymbolDeclarationAsync(targetDoc, targetSource, symbolRange, statusBar, cancellationToken)
+            }
+            |> Async.map (fun a -> a.IsSome)
+            |> RoslynHelpers.StartAsyncAsTask cancellationToken
+
+[<Export(typeof<IFSharpCrossLanguageSymbolNavigationService>)>]
+[<Export(typeof<FSharpCrossLanguageSymbolNavigationService>)>]
+type FSharpCrossLanguageSymbolNavigationService()  =
+    let componentModel = Package.GetGlobalService(typeof<ComponentModelHost.SComponentModel>) :?> ComponentModelHost.IComponentModel
+    let workspace = componentModel.GetService<VisualStudioWorkspace>()
+    let statusBar = StatusBar(ServiceProvider.GlobalProvider.GetService<SVsStatusbar,IVsStatusbar>())
+    let metadataAsSource = componentModel.DefaultExportProvider.GetExport<FSharpMetadataAsSourceService>().Value
+
+    // So, the groups are following:
+    //   1 - type (see below).
+    //   2 - Path - a dotted path to a symbol.
+    //   3 - parameters, opetional, only for methods and properties.
+    //   4 - return type, optional, only for methods.
+    let docCommentIdRx = Regex(@"^(\w):([\w\d#`.]+)(\(.+\))?(?:~([\w\d.]+))?$", RegexOptions.Compiled)
```

Comment:
Nothing can go wrong with regexes, amirite?

Reply:
That is clearly bullet proof.